classes/AjaxResponse.php

```diff
@@ -110,81 +110,79 @@ public function getResponse()
                 ]
             ),
         ];
-
-        return $return;
     }
 
     /**
      * @return string Json encoded response from $this->getResponse()
      */
-    public function getJson()
+    public function getJson(): string
     {
         return json_encode($this->getResponse());
     }
 
     // GETTERS
 
-    public function getError()
+    public function getError(): ?bool
     {
         return $this->error;
     }
 
-    public function getStepDone()
+    public function getStepDone(): ?bool
     {
         return $this->stepDone;
     }
 
-    public function getNext()
+    public function getNext(): ?string
     {
         return $this->next;
     }
 
-    public function getStatus()
+    public function getStatus(): string
     {
         return $this->getNext() == 'error' ? 'error' : 'ok';
     }
 
-    public function getNextParams()
+    public function getNextParams(): array
     {
         return $this->nextParams;
     }
 
-    public function getUpgradeConfiguration()
+    public function getUpgradeConfiguration(): UpgradeConfiguration
     {
         return $this->upgradeConfiguration;
     }
 
     // SETTERS
 
-    public function setError($error)
+    public function setError(?bool $error): AjaxResponse
```

Comment:
Basically I didn't have any nullables, but had setters with null values, which caused typing errors

Reply:
Could it be improved if we set the type to some variables?

Reply:
This might do yes, but still requires a lot of work. Like here https://github.com/PrestaShop/autoupgrade/blob/330ecdd950acc1a247d166a3630c940082030279/classes/TaskRunner/AbstractTask.php#L61 you must type error, know its type well and ensure that it does not have several. I think this could be done later.